src/PairEvaluatorColloid.h

```diff
@@ -255,42 +261,40 @@ class PairEvaluatorColloid : public PairEvaluator
         // compute the force divided by r in force_divr
         if (rsq < rcutsq && A != 0)
             {
-            if (form == SOLVENT_SOLVENT)
+            // Solvent-Solvent interaction
+            if (ai == 0 && aj == 0)
```

Comment:
Wrap all these zeros here & below in `Scalar` to make sure there's no type casting.
```suggestion
            if (ai == Scalar(0) && aj == Scalar(0))
```